drivers/pwm/pwm_mspm0.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2024 Bang & Olufsen A/S, Denmark
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ti/driverlib/dl_timera.h>
+#include <ti/driverlib/dl_timerg.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(pwm_mspm0, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define DT_DRV_COMPAT ti_mspm0_pwm
+
+struct pwm_mspm0_config {
+	GPTIMER_Regs *timer;
+	const struct pinctrl_dev_config *pinctrl;
+	const uint32_t frequency;
+	const DL_Timer_ClockConfig clock_cfg;
+	const DL_TIMER_PWM_MODE pwm_mode;
+	const uint8_t cc_index;
+	const uint8_t direction;
+	const bool is_tima;
+	const bool four_cc;
+};
+
+static int pwm_mspm0_get_cycles_per_sec(const struct device *dev, uint32_t channel,
+					uint64_t *cycles)
+{
+	const struct pwm_mspm0_config *config = dev->config;
+	*cycles = config->frequency /
+		  ((config->clock_cfg.divideRatio + 1) * (config->clock_cfg.prescale + 1));
+	return 0;
+}
+
+static int pwm_mspm0_set_cycles(const struct device *dev, uint32_t channel, uint32_t period_cycles,
+				uint32_t pulse_cycles, pwm_flags_t flags)
```

Comment:
There is probably _some_ expectation that the flags are used or at least checked for support. Otherwise clients are in for a nice surprise when all of their flags change nothing.
Also note that the first 8 bits are reserved for generic flags while the last 8 bits are reserved for soc specific ones. So your magical `1` to stop the pwm breaks the rules.

Reply:
fair point, should be addressed with the latest change.